pros/src/adi/analog.rs

```diff
@@ -0,0 +1,88 @@
+use crate::adi::{
+    AdiError,
+    port::AdiPort
+};
+
+use crate::error::bail_on;
+
+use pros_sys::{
+    E_ADI_ANALOG_IN,
+    E_ADI_ANALOG_OUT,
+    PROS_ERR
+};
+
+use core::ops::{
+    Deref,
+    DerefMut
+};
+
+pub struct AdiAnalogIn<'a> {
+    port: &'a AdiPort,
```

Comment:
I don't think that taking a reference to `AdiPort` is necessary in this case. Registering more than one device on a singular port index is unsound due to causing issues with the borrow checker. Once #24 is figured out fully we might have a clearer idea on how storing the port will work, but it might just be easier to take it by value for now.